deepchecks/nlp/checks/data_integrity/frequent_substrings.py

```diff
@@ -0,0 +1,417 @@
+            final_key = long_ngram
+            final_freq = results[long_ngram]['freq']
+            for ngram in ngram_strings:
+                if ngram not in results:
+                    continue
+                if ngram in final_key:
+                    if ngram == final_key:
+                        continue
+                    ngram_freq = results[ngram]['freq']
+                    if ngram_freq - final_freq <= self.frequency_margin:
+                        del results[ngram]
+                    else:
+                        del results[final_key]
+                        final_key = ngram
+                        final_freq = ngram_freq
+        return results
+
+    def _get_significant_cut_ind(self, df):
+        """
+        Determine the index cutoff for substrings with frequencies above a significant threshold.
+
+        This method identifies the position in the dataframe 'df' where the substring's frequency
+        surpasses the defined threshold `self.significant_substring_ratio`. If there are multiple
+        substrings meeting the criteria, it returns the index after the last such substring. If no
+        such substring exists, it returns -1.
+
+        Parameters:
+        -----------
+        df : pd.DataFrame
+            A sorted Dataframe containing substring information. Expected to have a 'Frequency' column
+            that lists the frequency of each substring in the dataset.
+
+        Returns:
+        --------
+        int
+            The index after the last substring that meets the significant frequency threshold or
+            -1 if no such substring exists.
+
+        Notes:
+        ------
+        The method is useful for filtering out substrings below a certain significance level based
+        on their frequencies.
+        """
+        significant_df = df[df['Frequency'] >= self.significant_substring_ratio]
+        if len(significant_df) > 0:
+            return list(significant_df.index)[-1] + 1
+        return -1
+
+    def _identify_peak_cut(self, df):
+        """
+        Identify the index at which there's a maximum relative change (peak) in the 'Frequency' column.
+
+        The function calculates the absolute difference between consecutive values in the 'Frequency' column
+        and divides this by the previous row's value to find the relative change or ratio. If the relative
+        change is below a certain threshold (`self.min_relative_change`), it is set to zero. The index of the maximum
+        relative change (peak) is returned.
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            Input dataframe with a 'Frequency' column.
+
+        Returns
+        -------
+        int
+            The index of the maximum relative change in the 'Frequency' column. If no significant peak is
+            found, returns the length of the dataframe.
+        """
+        diff = df['Frequency'].diff().abs()
+        ratio = diff.div(df['Frequency'].shift(1))
```

Comment:
sure .shift(1) is the current value and not the next value? 
To rephrase, asking if it's (t[i]-t[i-1]) / t[i] or (t[i]-t[i-1]) / t[i-1]. I think that it should be the first one.